docs/design/interface/pseudocode/write_resource_data_to_raw.py

```diff
@@ -0,0 +1,120 @@
+    [design](https://sprout.seedcase-project.org/docs/design/) docs for an
+    explanation of this file. Use `path_resource_raw()` to provide the
+    correct `path` location. Copies and compresses the file, and outputs the
+    path object of the created file.
+
+        Args:
+            data_path: Path to a raw data file that you want Sprout to store
+                into the data package.
+            path: Path to the raw resources folder. Use `path_resources_raw()`
+                to help provide the correct path.
+            resource_properties: The properties object for the specific resource.
+                Use `read_properties()` to read the properties for the resource
+                and `get_properties()` to get the correct resource properties.
+
+        Returns:
+            The path to the written compressed raw resource file.
+
+        Examples:
+            ```{python}
+            #| eval: false
+            ```
+    """
+    check_is_dir(path)
+    check_is_file(data_path)
+    check_is_supported_format(data_path)
+    check_data_basics(data_path, resource_properties)
+    check_data_constraints(data_path, resource_properties)
+
+    raw_resource_path = Path(path / create_raw_file_name(data_path))
+    return write_compressed_file(data_path, raw_resource_path)
+
+
+def create_raw_file_name(data_path: Path) -> Error | None:
+    """Creates a timestamped, unique file name for the raw data file.
+
+    This function creates a timestamped, unique file name for the raw data file
+    that is being copied into the resource's raw data folder. This file name
+    is used to store the raw data file as a backup. The file name will be
+    in the format `{timestamp}-{uuid}.{extension}.gz`, where `timestamp` is the
+    current time following ISO8601 format, and `uuid` is a universally unique ID.
+
+        Args:
+            data_path: The path to the raw data file.
+
+        Returns:
+            A path to a (potentially) new raw resource file.
+    """
+    # Untested code.
+    timestamp = datetime.datetime.now().isoformat()
+    uuid = str(uuid.uuid4())
+    extension = data_path.suffix
+    return Path("{timestamp}-{uuid}.{extension}.gz")
+
+
+def write_compressed_file(data_path: Path, path: Path) -> Path:
+    """Write and compress the raw data file into the resource's raw data folder.
+
+    This function copies the raw data file from `data_path` to the resource's
+    raw data folder, compressing it in the process. The compressed file is
+    stored in the raw data folder based on the path provided.
+
+        Args:
+            data_path: The path to the raw data file.
+            path: The path to the raw resource file in the
+                resource's raw data folder.
+
+        Returns:
+            The path to the written compressed raw resource file.
+    """
+    return path
+
+
+def check_data_basics(
+    data_path: Path, resource_properties: ResourceProperties
+) -> Error | None:
```

Comment:
```suggestion
) -> Path:
```
To make it like the other check functions, it could return its input if the checks passed and throw an error if any failed.